src/components/WithClickOutside.js

```diff
@@ -1,21 +1,33 @@
-import React, { useState, useRef, useEffect } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { useLocation } from 'react-router-dom';
 
 const WithClickOutside = (WrappedComponent) => {
   const Component = () => {
+    let location = useLocation();
     const [open, setOpen] = useState(false);
-
-    const ref = useRef();
+    const ref = useRef(null);
+    const setRef = useCallback((node) => {
```

Comment:
Here, try to reduce the complexity

Reply:
Managed from 11 to 8
I will see different ways to improve the component but not today. Maybe in a couple of hours. Will sent another PR in case this is fixed up. @samuel-shyaka-dus I could use some ideas.

Reply:
The complexity of the code has been resolved successfully @samuel-shyaka-dus